packages-exp/remote-config-exp/src/api.ts

```diff
@@ -0,0 +1,176 @@
+  // * it is consistent with the Fetch API's signal input
+  const abortSignal = new RemoteConfigAbortSignal();
+
+  setTimeout(async () => {
+    // Note a very low delay, eg < 10ms, can elapse before listeners are initialized.
+    abortSignal.abort();
+  }, rc.settings.fetchTimeoutMillis);
+
+  // Catches *all* errors thrown by client so status can be set consistently.
+  try {
+    await rc._client.fetch({
+      cacheMaxAgeMillis: rc.settings.minimumFetchIntervalMillis,
+      signal: abortSignal
+    });
+
+    await rc._storageCache.setLastFetchStatus('success');
+  } catch (e) {
+    const lastFetchStatus = hasErrorCode(e, ErrorCode.FETCH_THROTTLE)
+      ? 'throttle'
+      : 'failure';
+    await rc._storageCache.setLastFetchStatus(lastFetchStatus);
+    throw e;
+  }
+}
+
+export function getAll(remoteConfig: RemoteConfig): Record<string, ValueType> {
+  const rc = asRemoteConfigImpl(remoteConfig);
+  return getAllKeys(
+    rc._storageCache.getActiveConfig(),
+    rc.defaultConfig
+  ).reduce((allConfigs, key) => {
+    allConfigs[key] = getValue(remoteConfig, key);
+    return allConfigs;
+  }, {} as Record<string, ValueType>);
+}
+
+export function getBoolean(remoteConfig: RemoteConfig, key: string): boolean {
+  return getValue(remoteConfig, key).asBoolean();
+}
+
+export function getNumber(remoteConfig: RemoteConfig, key: string): number {
+  return getValue(remoteConfig, key).asNumber();
+}
+
+export function getString(remoteConfig: RemoteConfig, key: string): string {
+  return getValue(remoteConfig, key).asString();
+}
+
+export function getValue(remoteConfig: RemoteConfig, key: string): ValueType {
+  const rc = asRemoteConfigImpl(remoteConfig);
+  if (!rc._isInitializationComplete) {
+    rc._logger.debug(
+      `A value was requested for key "${key}" before SDK initialization completed.` +
+        ' Await on ensureInitialized if the intent was to get a previously activated value.'
+    );
+  }
+  const activeConfig = rc._storageCache.getActiveConfig();
+  if (activeConfig && activeConfig[key] !== undefined) {
+    return new Value('remote', activeConfig[key]);
+  } else if (rc.defaultConfig && rc.defaultConfig[key] !== undefined) {
+    return new Value('default', String(rc.defaultConfig[key]));
+  }
+  rc._logger.debug(
+    `Returning static value for key "${key}".` +
+      ' Define a default or remote value if this is unintentional.'
+  );
+  return new Value('static');
+}
+
+export function setLogLevel(
+  remoteConfig: RemoteConfig,
+  logLevel: RemoteConfigLogLevel
+): void {
+  const rc = asRemoteConfigImpl(remoteConfig);
+  switch (logLevel) {
+    case 'debug':
+      rc._logger.logLevel = FirebaseLogLevel.DEBUG;
+      break;
+    case 'silent':
+      rc._logger.logLevel = FirebaseLogLevel.SILENT;
+      break;
+    default:
+      rc._logger.logLevel = FirebaseLogLevel.ERROR;
+  }
+}
+
+function asRemoteConfigImpl(remoteConfig: RemoteConfig): RemoteConfigImpl {
+  return remoteConfig as RemoteConfigImpl;
```

Comment:
Minor: I'm guessing it's a best-practice to wrap the cast operation, but it's new to me. Is there a doc you'd recommend for the rationale?

Reply:
Yeah, I don't think it's a best-practice. I didn't want to do casting(using `as`) in every API, so I moved it to a function. On reflection, I think it's probably not necessary, and will waste cycles by introducing a no-op function call. 

Changed to do casting directly in individual functions.